### PR TITLE
Javascript payment renderer refactoring

### PR DESCRIPTION
**Pull request by jacstn**
#### 1. Objective

Small JS renderer refactoring,
Reason of this PR is move some common, repeated functionality to separated function.

#### 2. Description of change

Java Script Renderer files, literally cut few lines of code and paste it to separated function.

#### 3. Quality assurance

**🔧 Environments:**

- **Platform version**: Magento CE 2.2.5.
- **Omise plugin version**: Omise-Magento 2.3.
- **PHP version**: 7.0.29.

**✏️ Details:**

To properly test it make test payments by Alipay Internet Banking, and Credit Card: both successful and unsuccessful. Also in scenarios when there is no internet connection.
 
#### 4. Impact of the change

None

#### 5. Priority of change

Normal

#### 6. Additional Notes

None

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js [hunk 1]
```diff
@@ -152,9 +161,7 @@ define(
                         self.getPlaceOrderDeferredObject()
                             .fail(
```
Comment (jonrandy) [inline view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js]: `.fail(self.handlePaymentError)` is all you need here

File: view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js [hunk 2]
```diff
@@ -169,19 +176,15 @@ define(
                                         storage.get(serviceUrl, false)
                                             .fail(
                                                 function (response) {
-                                                    errorProcessor.process(response, self.messageContainer);
-                                                    fullScreenLoader.stopLoader();
-                                                    self.isPlaceOrderActionAllowed(true);
+                                                    self.handlePaymentError(response);
```
Comment (jonrandy) [inline view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js]: Same here

File: view/frontend/web/js/view/payment/method-renderer/omise-offsite-alipay-method.js [hunk 3]
```diff
@@ -82,9 +89,7 @@ define(
                 self.getPlaceOrderDeferredObject()
                     .fail(
                         function (response) {
-                            errorProcessor.process(response, self.messageContainer);
-                            fullScreenLoader.stopLoader();
-                            self.isPlaceOrderActionAllowed(true);
+                            self.handlePaymentError(response);
```
Comment (jonrandy) [inline view/frontend/web/js/view/payment/method-renderer/omise-offsite-alipay-method.js]: same again

File: view/frontend/web/js/view/payment/method-renderer/omise-offsite-alipay-method.js [hunk 4]
```diff
@@ -100,19 +105,15 @@ define(
                             storage.get(serviceUrl, false)
                                 .fail(
                                     function (response) {
-                                        errorProcessor.process(response, self.messageContainer);
-                                        fullScreenLoader.stopLoader();
-                                        self.isPlaceOrderActionAllowed(true);
+                                        self.handlePaymentError(response);
```
Comment (jonrandy) [inline view/frontend/web/js/view/payment/method-renderer/omise-offsite-alipay-method.js]: and again

File: view/frontend/web/js/view/payment/method-renderer/omise-offsite-internetbanking-method.js [hunk 5]
```diff
@@ -92,9 +101,7 @@ define(
                 self.getPlaceOrderDeferredObject()
                     .fail(
                         function(response) {
-                            errorProcessor.process(response, self.messageContainer);
-                            fullScreenLoader.stopLoader();
-                            self.isPlaceOrderActionAllowed(true);
+                            self.handlePaymentError(response);
```
Comment (jonrandy) [inline view/frontend/web/js/view/payment/method-renderer/omise-offsite-internetbanking-method.js]: I probably don't need to say more

File: view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js (hunk 1 above)
Comment (jacekstan) [reply view/frontend/web/js/view/payment/method-renderer/omise-cc-method.js]: I was thinking about that.. and finally no... i don't like it.. as we are loosing function argument.
I prefere to see that `response`..

File: view/frontend/web/js/view/payment/method-renderer/omise-offsite-internetbanking-method.js [hunk 6]
```diff
@@ -108,21 +113,13 @@ define(
...
 
                             storage.get(serviceUrl, false)
-                                .fail(
-                                    function (response) {
-                                        errorProcessor.process(response, self.messageContainer);
-                                        fullScreenLoader.stopLoader();
-                                        self.isPlaceOrderActionAllowed(true);
-                                    }
-                                )
+                                .fail(response => elf.handlePaymentError(self, response))
```
Comment (guzzilar) [inline view/frontend/web/js/view/payment/method-renderer/omise-offsite-internetbanking-method.js]: @jacstn `elf` > `self`?

**Pull request closed without merging**